packages/gatsby-plugin-postcss/src/gatsby-node.js

```diff
@@ -0,0 +1,90 @@
+import resolve from "./resolve"
+
+const CSS_PATTERN = /\.css$/
+const MODULE_CSS_PATTERN = /\.module\.css$/
+
+const isBuiltInCssRule = rule =>
+  rule.test &&
+  (rule.test.toString() === CSS_PATTERN.toString() ||
+    rule.test.toString() === MODULE_CSS_PATTERN.toString())
+
+const getOptions = pluginOptions => {
+  const options = { ...pluginOptions }
+
+  delete options.plugins
+
+  const postcssPlugins = options.postCssPlugins
+
+  if (postcssPlugins) {
+    options.plugins = postcssPlugins
+  }
+
+  delete options.postCssPlugins
+
+  return options
+}
+
+const removeBuiltInCssLoaders = config => {
```

Comment:
There shouldn't be a need for this. This plugin will already intercept all css files and run the through producing the extracting file or tye inlined style before the built-in rule can run. I'd say trying to not mess with the config is safer long term and in more use cases than  effectively guessing which loader is the built in one

Reply:
@jquense I completely agree with you.

Unfortunately, if we have i.e. two rules with the same `test` matchers they will be applied in the order they are defined in `module.rules`:

```js
module: {
  rules: [{
      test: /\.css$/,
      use: []
    }, {
      test: /\.css$/,
      use: []
  }]
}
```

If we don't remove we will get:

```
Error: ./src/components/layout/index.css
  Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
  ModuleBuildError: Module build failed (from ./node_modules/mini-css-extract-plugin/dist/l  oader.js):
  Error: Didn't get a result from child compiler
```

Now we have a couple of options:

1. Remove existing `css` rules manually
2. Remove existing `css` rules from the core completely
2. Remove existing `css` rules from the core and move their defaults into that plugin

Reply:
Hmm loaders shouldn't be run in order of defining, but in reverse order (see  https://stackoverflow.com/questions/32234329/what-is-the-loader-order-for-webpack) I'd like to take a moment to understand what's happening let me try and do some debugging

Reply:
It looks like Gatsby is always using `oneOf` for handling CSS, to prevent `*.module.css` files from matching both module and non-module CSS rules. Would it be possible to prepend these new rules before the existing ones? The end result would look like:
```
oneOf: [
  postcssModuleRule,
  postcssRule,
  rules.cssModules(), 
  rules.css()
]
```
This way, the default rules would never match anyway, because the prepended ones use the same regexes.

Reply:
@szimek I tried your approach and it worked fine. Thanks! I will publish an update.

Reply:
@jquense What do you think about an approach suggested by @szimek? I have already used it and it would be nice if you could review it.

Reply:
I think this is a good approach for now 👍